addon/mixins/dom.js

```diff
@@ -103,76 +70,47 @@ export default Mixin.create({
       this.isComponent || typeof selector !== 'string'
     );
 
-    let element = findElement(this.element, selector);
-    let callback = run.bind(this, _callback);
+    let element;
 
-    if (this._listeners === undefined) {
-      this._listeners = [];
+    // If no element is provided, we assume we're adding the event listener to the component's element. This
+    // addresses use cases where we want to bind events like `scroll` to the component's root element.
+    if (this.isComponent && typeof eventName === 'function') {
```

Comment:
seems like this conditional does not take care of `options`, am I missing something?

Specifically if someone using the mixin does:

```
this.addEventListener(‘fooEvent’, function(){}, { passive: true });
```

Reply:
Doh. Good catch.